### PR TITLE
Add recipe api interface

### PR DESCRIPTION
**Pull request by mplekunov**

### CODE REVIEW

File: server/recipeAPI/IRecipeAPI.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+     * 
+     * @param parameters parameters used for searching.
+     * - query - required parameter that defines the name of the Recipe item (partial names are accepted).
+     * - number - optional parameter that defines max number of the results to be returned.
+     * - page - optional parameter that defines page number for pagination.
+     * - intolerence - optional parameter that defines the type of intolerences to be taken in consideration during searching.
+     * 
+     * @throws NoParameterFound exception when required parameters weren't found.
+     *
+     * @returns Promise filled with a collection of Partial<IRecipe> objects.
```
Comment (nashirj) [inline server/recipeAPI/IRecipeAPI.ts:15]: Spacing between param throw and returns is different from in SpoonacularAPI for example, same below

File: server/recipeAPI/IRecipeAPI.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import IRecipe from "../serverAPI/model/recipe/IRecipe";
+
+export default interface IRecipeAPI {
+     /**
+     * Retrieves collection of partially filled Recipe items that satisfy searching parameters.
+     * 
+     * @param parameters parameters used for searching.
+     * - query - required parameter that defines the name of the Recipe item (partial names are accepted).
```
Comment (nashirj) [inline server/recipeAPI/IRecipeAPI.ts:8]: Maybe `recipeName` is more descriptive?

File: server/recipeAPI/IRecipeAPI.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+     * - number - optional parameter that defines max number of the results to be returned.
+     * - page - optional parameter that defines page number for pagination.
+     * - intolerence - optional parameter that defines the type of intolerences to be taken in consideration during searching.
```
Comment (nashirj) [inline server/recipeAPI/IRecipeAPI.ts:11]: Typo: `intolerance`

File: server/recipeAPI/IRecipeAPI.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import IRecipe from "../serverAPI/model/recipe/IRecipe";
+
+export default interface IRecipeAPI {
+     /**
+     * Retrieves collection of partially filled Recipe items that satisfy searching parameters.
+     * 
+     * @param parameters parameters used for searching.
+     * - query - required parameter that defines the name of the Recipe item (partial names are accepted).
+     * - number - optional parameter that defines max number of the results to be returned.
+     * - page - optional parameter that defines page number for pagination.
```
Comment (nashirj) [inline server/recipeAPI/IRecipeAPI.ts:10]: For all optional parameters, please specify what the default value is

File: server/recipeAPI/IRecipeAPI.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+     *
+     * @returns Promise filled with a collection of Partial<IRecipe> objects.
+     */
+    SearchRecipe(parameters: Map<string, any>): Promise<Partial<IRecipe>[]>;
+
+    /**
+     * Retrieves specific Recipe item that is defined by unique identifier.
+     * 
+     * @param parameters parameters used for searching.
+     * - id - required parameter that defines unique identifier of the Recipe item
```
Comment (nashirj) [inline server/recipeAPI/IRecipeAPI.ts:23]: period

**Review by nashirj — APPROVED** [review]
Small nit comments but lgtm

File: server/recipeAPI/IRecipeAPI.ts (hunk 4 above)
Comment (mplekunov) [reply server/recipeAPI/IRecipeAPI.ts:10]: well, default is non... I just don't use them if they are not specified

**Pull request merged**